ckanext/unaids/tests/test_plugin.py

```diff
@@ -30,3 +33,61 @@ def test_geojson_format_guessed_correctly(self):
         response = call_action('resource_create', {}, **resource)
         response = call_action('package_show', {}, id=dataset['id'])
         assert response['resources'][0]['format'] == 'GeoJSON'
+
+
+class TestResourceLastModified(object):
+    '''Tests for the ckanext.plugin._update_resource_last_modified_date module.
+
+    Make sure we are setting the last_modified to the resource dict
+    whenever a resource is created or edited.
+    '''
+
+    URL_TYPE_FILE_UPLOAD = 'upload'
+    URL_TYPE_LINK = ''
+
+    def test_resource(self, url_type):
+        resource_dict = {'url_type': url_type}
+        if url_type == self.URL_TYPE_FILE_UPLOAD:
+            resource_dict.update({
+                'url': 'file.csv',
+                'lfs_prefix': 'prefix123',
+                'sha256': 'sha256',
+                'size': 100,
+            })
+        elif url_type == self.URL_TYPE_LINK:
+            resource_dict.update({
+                'url': 'http://example.com'
+            })
+        return resource_dict
```

Comment:
This should be two fixtures: 
https://docs.pytest.org/en/6.2.x/fixture.html#what-fixtures-are
1.
```
@pytest.fixture
def resource_with_file():
[...]
```
```
@pytest.fixture
def resource_with_link():
[...]
```


Then:
1.
```
    def test_null_to_file(self, resource_with_file):
        _update_resource_last_modified_date(resource=resource_with_file)
        assert 'last_modified' in resource
```
2.
```
    def test_file_to_url(self, resource_with_file, resource_with_link):
        current = self.test_resource(self.URL_TYPE_FILE_UPLOAD)
        resource = self.test_resource(self.URL_TYPE_LINK)
        _update_resource_last_modified_date(resource_with_link, current=resource_with_file)
        assert 'last_modified' in resource
```
etc.